### PR TITLE
tests: more backtesting testing

### PR DESCRIPTION
**Pull request by kryofly**

### CODE REVIEW

File: freqtrade/tests/test_analyze.py [hunk 1]
```diff
@@ -19,8 +19,11 @@ def test_dataframe_correct_columns(result):
 
 
 def test_dataframe_correct_length(result):
-    # no idea what this check truly does - should we just remove it?
-    assert len(result.index) == 14397
+    # FIX: no idea what this check truly does - should we just remove it?
+    # magic needs to change everytime testdata/* is updated.
+    # Perhaps change result fixture to use BTC_UNITEST instead?
```
[the remark this answers is not in the document]
Comment (kryofly) [reply freqtrade/tests/test_analyze.py]: Wasn't sure why that fixture isn't using BTC_UNITEST, perhaps it was made before the testpair was created.

File: freqtrade/tests/optimize/test_backtesting.py [hunk 2]
```diff
@@ -129,17 +194,92 @@ def simple_backtest(config, contour, num_results):
 
 # Test backtest on offline data
 # loaded by freqdata/optimize/__init__.py::load_data()
+# Note alot of tests has:
```
[the remark this answers is not in the document]
Comment (kryofly) [reply freqtrade/tests/optimize/test_backtesting.py]: I see them if using pytest -s, its unrelated to this PR. Not sure if its real errors, perhaps some fixture that is using the strategy in a wrong way.

File: freqtrade/tests/optimize/test_backtesting.py [hunk 3]
```diff
@@ -18,6 +19,70 @@ def trim_dictlist(dict_list, num):
...
+            buy[i] = 1
+        else:
+            sell[i] = 1
+    signals['buy'] = buy
+    signals['sell'] = sell
+    return dataframe
+
+
+def _run_backtest_1(strategy, fun, backtest_conf):
+    # strategy is a global (hidden as a singleton), so we
```
[the remark this answers is not in the document]
Comment (kryofly) [reply freqtrade/tests/optimize/test_backtesting.py]: Yes, also perhaps more pure functions. The amount of mocking makes it hard to both write and understand test case. Perhaps make them more fragile too.

[the remark this answers is not in the document]
Comment (vertti) [reply freqtrade/tests/optimize/test_backtesting.py]: True.

File: freqtrade/tests/test_analyze.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kryofly) [reply freqtrade/tests/test_analyze.py]: Should this test be rewritten into something that uses load_data to get the tickerhistory structure (loaded from BTC_UNITEST), and then calls preprocess to also get a dataframe, and finally compare the size of them two?

[the remark this answers is not in the document]
Comment (vertti) [reply freqtrade/tests/test_analyze.py]: That might make it more valuable.

File: freqtrade/tests/optimize/test_backtesting.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vertti) [reply freqtrade/tests/optimize/test_backtesting.py]: Ok. So let's fix those in another PR. But rather make a new issue to github than writing those kind of commnents to the code.

**Comment by vertti** [comment]
Thanks!

**Pull request merged**